gix-blame/src/file/tests.rs

```diff
@@ -918,368 +870,345 @@ mod process_change {
         let hunks_to_blame = vec![new_unblamed_hunk(0..6, suspect, Offset::Added(0))];
         let changes = vec![Change::AddedOrReplaced(0..4, 3), Change::Unchanged(4..6)];
-        let new_hunks_to_blame = process_changes(&mut lines_blamed, hunks_to_blame, changes, suspect);
+        let new_hunks_to_blame = process_changes(hunks_to_blame, changes, suspect, parent);
 
         assert_eq!(
-            lines_blamed,
-            [BlameEntry {
-                start_in_blamed_file: 0,
-                start_in_source_file: 0,
-                len: force_non_zero(4),
-                commit_id: suspect
-            }]
+            new_hunks_to_blame,
+            [
+                UnblamedHunk {
+                    range_in_blamed_file: 0..4,
+                    suspects: [(suspect, 0..4)].into(),
+                },
+                UnblamedHunk {
+                    range_in_blamed_file: 4..6,
+                    suspects: [(suspect, 4..6), (parent, 3..5)].into(),
+                }
+            ]
         );
-        assert_eq!(new_hunks_to_blame, [new_unblamed_hunk(4..6, suspect, Offset::Added(1))]);
     }
 
     #[test]
     fn added_hunk_6() {
-        let mut lines_blamed = Vec::new();
-        let suspect = ObjectId::null(gix_hash::Kind::Sha1);
+        let suspect = ObjectId::from_str("0000000000000000000000000000000000000000").unwrap();
+        let parent = ObjectId::from_str("1111111111111111111111111111111111111111").unwrap();
         let hunks_to_blame = vec![new_unblamed_hunk(4..6, suspect, Offset::Added(1))];
         let changes = vec![Change::AddedOrReplaced(0..3, 0), Change::Unchanged(3..5)];
-        let new_hunks_to_blame = process_changes(&mut lines_blamed, hunks_to_blame, changes, suspect);
+        let new_hunks_to_blame = process_changes(hunks_to_blame, changes, suspect, parent);
 
-        assert_eq!(lines_blamed, []);
-        assert_eq!(new_hunks_to_blame, [new_unblamed_hunk(4..6, suspect, Offset::Added(4))]);
+        assert_eq!(
+            new_hunks_to_blame,
+            [UnblamedHunk {
+                range_in_blamed_file: 4..6,
+                suspects: [(suspect, 3..5), (parent, 0..2)].into(),
+            }]
+        );
     }
 
     #[test]
     fn added_hunk_7() {
-        let suspect = ObjectId::null(gix_hash::Kind::Sha1);
-        let suspect_2 = ObjectId::from_hex(b"2222222222222222222222222222222222222222").unwrap();
-        let mut lines_blamed: Vec<BlameEntry> = vec![BlameEntry {
-            start_in_blamed_file: 0,
-            start_in_source_file: 0,
-            len: force_non_zero(1),
-            commit_id: suspect,
-        }];
-        let hunks_to_blame = vec![new_unblamed_hunk(1..3, suspect_2, Offset::Added(1))];
+        let suspect = ObjectId::from_hex(b"0000000000000000000000000000000000000000").unwrap();
+        let parent = ObjectId::from_str("1111111111111111111111111111111111111111").unwrap();
```

Comment:
Here I thought that a constant would be much less noisy and not harder to read, but I leave it to your preference and maybe a next PR can adjust this if you agree.

Reply:
I just tried, and it appears that we can’t extract into a constant as neither `from_hex` nor `from_str` work in a `const` context which would be required for `const SUSPECT: ObjectId = …`. Or did you have something else in mind?

Reply:
Right, these aren't const so can't easily be used in a static context. I think then I would define functions for it that deal with the `.unwrap()`, so it looks more like `let suspsect = null_sha(); let parent = one_sha()`.

Reply:
Makes sense! Done in #1756.